### PR TITLE
feat(devoir) :EVAL-399 import notes with the exercozer csv file part front

### PR DESCRIPTION
**Pull request by lapeyronnie**
## Describe your changes
ajout d'un bouton et une popup pour import un fichier csv
[EVAL-45](https://jira.support-ent.fr/browse/EVAL-45)
## Issue ticket number and link
[EVAL-399](https://jira.support-ent.fr/browse/EVAL-399)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/directives/import-note/import-note.directive.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            this.files[0].name.endsWith('.csv') && this.files[0].name.trim() !== '' : false ;
+    }
+}
+
+function directive() {
+    return {
+        restrict: 'E',
+        templateUrl: `${RootsConst.directive}import-note/import-note.html`,
+        scope: {
+            //onImport: "&",
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/directives/import-note/import-note.directive.ts:120]: supprime ce commentaire

File: src/main/resources/public/ts/directives/import-note/import-note.directive.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    openLightboxImportNote: boolean;
+    isErrorStudent: boolean;
+    files: File[];
+    devoir: Devoir;
+    multiple: boolean;
+    errorMessage: string;
+    lang: typeof idiom;
+
+    constructor(private $scope: IImportScope,
+                private $location:ILocationService,
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/directives/import-note/import-note.directive.ts:48]: tu n'as pas l'air de te servir de ton location, tu peux faire du nettoyage du coup 😄

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/directives/import-note/import-note.directive.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    disabledButton(): boolean {
+        return this.files.length === 0 || (this.errorMessage && this.errorMessage.length > 0);
+    }
+
+    async importNote() : Promise<any> {
+            delete this.errorMessage;
+            let formData = new FormData();
+            formData.append('file', this.files[0], this.files[0].name);
+
+            await NoteService.importNote(this.devoir.id_groupe, this.devoir.id)
```
Comment (sjollois) [inline src/main/resources/public/ts/directives/import-note/import-note.directive.ts:93]: tu ne mets pas ton formData dedans ?

**Review by sjollois — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/directives/import-note/import-note.directive.ts (hunk 2 above)
Comment (lapeyronnie) [reply src/main/resources/public/ts/directives/import-note/import-note.directive.ts:48]: pour l'instant

File: src/main/resources/public/ts/directives/import-note/import-note.directive.ts (hunk 3 above)
Comment (lapeyronnie) [reply src/main/resources/public/ts/directives/import-note/import-note.directive.ts]: bien vu

**Review by qpic-cgi — APPROVED** [review]

**Review by sjollois — APPROVED** [review]

**Pull request merged**